crates/pixi_build_frontend/src/tool/cache.rs

```diff
@@ -1,38 +1,149 @@
-use std::path::PathBuf;
+use std::{fmt::Debug, hash::Hash, path::PathBuf};
 
 use dashmap::{DashMap, Entry};
+use pixi_consts::consts::CONDA_REPODATA_CACHE_DIR;
+use rattler_conda_types::Channel;
+use rattler_repodata_gateway::{ChannelConfig, Gateway};
+use reqwest_middleware::ClientWithMiddleware;
 
 use super::IsolatedTool;
 use crate::{
     tool::{SystemTool, Tool, ToolSpec},
     IsolatedToolSpec, SystemToolSpec,
 };
 
+pub struct ToolContextBuilder {
+    gateway_config: ChannelConfig,
```

Comment:
It would be more ideal to simply pass in the gateway here! We want to be able to share the gateway with pixi itself because it is also fetching a lot of repodata.